core/client/fs/src/main/java/alluxio/client/file/UfsBaseFileSystem.java

```diff
@@ -0,0 +1,425 @@
+    Preconditions.checkArgument(!ufsAddress.isEmpty(), "ufs address should not be empty");
+    mRootUFS = new AlluxioURI(ufsAddress);
+    UfsManager.UfsClient ufsClient = new UfsManager.UfsClient(
+        () -> UnderFileSystem.Factory.create(ufsAddress, mFsContext.getClusterConf()),
+        new AlluxioURI(ufsAddress));
+    mUfs = ufsClient.acquireUfsResource();
+    mCloser.register(mFsContext);
+    mCloser.register(mUfs);
+    LOG.debug("Creating file system connecting to ufs address {}", ufsAddress);
+  }
+
+  /**
+   * Shuts down the FileSystem. Closes all thread pools and resources used to perform operations. If
+   * any operations are called after closing the context the behavior is undefined.
+   */
+  @Override
+  public synchronized void close() throws IOException {
+    if (!mClosed) {
+      mClosed = true;
+    }
+    mCloser.close();
+  }
+
+  @Override
+  public synchronized boolean isClosed() {
+    return mClosed;
+  }
+
+  @Override
+  public void checkAccess(AlluxioURI path, CheckAccessPOptions options) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void createDirectory(AlluxioURI path, CreateDirectoryPOptions options) {
+    call(() -> {
+      // TODO(lu) deal with other options e.g. owner/group
+      MkdirsOptions ufsOptions = MkdirsOptions.defaults(mFsContext.getPathConf(path));
+      if (options.hasMode()) {
+        ufsOptions.setMode(Mode.fromProto(options.getMode()));
+      }
+      if (options.hasRecursive()) {
+        ufsOptions.setCreateParent(options.getRecursive());
+      }
+      mUfs.get().mkdirs(path.getPath(), ufsOptions);
+    });
+  }
+
+  @Override
+  public FileOutStream createFile(AlluxioURI path, CreateFilePOptions options) {
+    return callWithReturn(() -> {
+      // TODO(lu) deal with other options e.g. owner/group/acl/ensureAtomic
+      CreateOptions ufsOptions = CreateOptions.defaults(mFsContext.getPathConf(path));
+      if (options.hasMode()) {
+        ufsOptions.setMode(Mode.fromProto(options.getMode()));
+      }
+      if (options.hasRecursive()) {
+        ufsOptions.setCreateParent(options.getRecursive());
+      }
+      return new UfsFileOutStream(mUfs.get().create(path.getPath(), ufsOptions));
+    });
+  }
+
+  @Override
+  public void delete(AlluxioURI path, DeletePOptions options) {
+    call(() -> {
+      String ufsPath = path.getPath();
+      if (mUfs.get().isFile(ufsPath)) {
+        mUfs.get().deleteFile(ufsPath);
+        return;
+      }
+      DeleteOptions ufsOptions = DeleteOptions.defaults();
+      if (options.hasRecursive()) {
+        ufsOptions.setRecursive(options.getRecursive());
+      }
+      mUfs.get().deleteDirectory(ufsPath, ufsOptions);
+    });
+  }
+
+  @Override
+  public boolean exists(AlluxioURI path, final ExistsPOptions options) {
+    return Boolean.TRUE.equals(callWithReturn(() -> mUfs.get().exists(path.getPath())));
```

Comment:
because the `callWithReturn` returns object instead of primitive types.
So it would return Boolean, already we know Boolean here will not be null, but IDE will show errors that unsafe unboxing